source/common/config/delta_subscription_impl.h

```diff
@@ -48,51 +41,42 @@ class DeltaSubscriptionImpl : public Subscription,
     request_.mutable_node()->MergeFrom(local_info_.node());
   }
 
-  // Enqueues and attempts to send a discovery request, (un)subscribing to resources missing from /
-  // added to the passed 'resources' argument, relative to resource_versions_.
-  void buildAndQueueDiscoveryRequest(const std::vector<std::string>& resources) {
-    ResourceNameDiff diff;
-    std::set_difference(resources.begin(), resources.end(), resource_names_.begin(),
-                        resource_names_.end(), std::inserter(diff.added_, diff.added_.begin()));
-    std::set_difference(resource_names_.begin(), resource_names_.end(), resources.begin(),
-                        resources.end(), std::inserter(diff.removed_, diff.removed_.begin()));
-
-    for (const auto& added : diff.added_) {
-      setResourceWaitingForServer(added);
+  void updateResources(std::vector<std::string> resource_names) override {
```

Comment:
I'm not sure what the calling code looks like, but would it be possible to have this parameter be a set instead? That way you'd avoid having to pass by value and sorting

Reply:
Definitely makes sense, and it doesn't look like any of the implementors would run into problems with this change. However, it is a pretty far reaching change, and there appears to be some tests I'll need to fix. Working on it.

Reply:
Done. This uncovered an issue with how the old non-delta gRPC xDS implementation uses its "watches". Previously, when updating your interest from resources 0 and 1 to just 2, you would first send a request for 0,1,2, and then send one for just 2. With my change, you'll now send a request for nothing, and then just 2. Well, I don't know, maybe this is too much churn; maybe the existing logic makes sense. Fortunately, this will all be replaced in the near future with logic that will have the correct behavior (just send one request, for 2).